medium-tests/src/test/java/mediumtest/fixtures/SonarLintBackendFixture.java

```diff
@@ -378,11 +379,66 @@ public SonarLintTestRpcServer build(SonarLintRpcClientDelegate client) {
     }
 
     private static SonarLintTestRpcServer createTestBackend(SonarLintRpcClientDelegate client) throws IOException {
-      var clientToServerOutputStream = new PipedOutputStream();
+      var clientToServerOutputStream = new PipedOutputStream() {
+        private final StringBuilder mem = new StringBuilder();
+
+        public int nextContentSize = -1;
+
+        @Override
+        public void write(@NotNull byte[] b) throws IOException {
+          mem.append(new String(b));
+          flushIfNeeded();
+          super.write(b);
+        }
+
+        public void flushIfNeeded() {
+          int cr = mem.indexOf("\r\n");
+          if (cr != -1 && nextContentSize < 0) {
+            var contentLength = mem.substring(0, cr);
+            mem.replace(0, cr + 2, "");
+            nextContentSize = Integer.parseInt(contentLength.substring("Content-Length: ".length()));
+          }
+          if (nextContentSize > 0 && mem.length() >= nextContentSize + 2) {
+            var content = mem.substring(2, nextContentSize + 2);
+            mem.replace(0, nextContentSize + 2, "");
+            nextContentSize = -1;
+            System.out.println("--> " + content);
+          }
+        }
+      };
       var clientToServerInputStream = new PipedInputStream(clientToServerOutputStream);
 
       var serverToClientOutputStream = new PipedOutputStream();
-      var serverToClientInputStream = new PipedInputStream(serverToClientOutputStream);
+      var serverToClientInputStream = new PipedInputStream(serverToClientOutputStream) {
+
+        private final StringBuilder mem = new StringBuilder();
+        public int nextContentSize = -1;
+
+        @Override
+        public synchronized int read(byte[] b, int off, int len) throws IOException {
+          int readLength = super.read(b, off, len);
+          if (readLength > 0) {
+            System.out.println("<-- " + new String(b, off, readLength));
+          }
+          return readLength;
+        }
+
+        public void flushIfNeeded() {
```

Comment:
Is this one used at all?

Reply:
Nope, this is a remain from a first attempt